src/panels/custom/ha-panel-custom.js

```diff
@@ -0,0 +1,119 @@
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import EventsMixin from '../../mixins/events-mixin.js';
+import NavigateMixin from '../../mixins/navigate-mixin.js';
+import loadCustomPanel from '../../util/custom-panel/load-custom-panel.js';
+import createCustomPanelElement from '../../util/custom-panel/create-custom-panel-element.js';
+import setCustomPanelProperties from '../../util/custom-panel/set-custom-panel-properties.js';
+
+/*
+ * Mixins are used by ifram to communicate with main frontend.
+ * @appliesMixin EventsMixin
+ * @appliesMixin NavigateMixin
+ */
+class HaPanelCustom extends NavigateMixin(EventsMixin(PolymerElement)) {
+  static get properties() {
+    return {
+      hass: Object,
+      narrow: Boolean,
+      showMenu: Boolean,
+      route: Object,
+      panel: {
+        type: Object,
+        observer: '_panelChanged',
+      }
+    };
+  }
+
+  static get observers() {
+    return [
+      '_dataChanged(hass, narrow, showMenu, route)'
+    ];
+  }
+
+  constructor() {
+    super();
+    this._setProperties = null;
+  }
+
+  _panelChanged(panel) {
+    // Clean up
+    delete window.customPanel;
+    this._setProperties = null;
+    while (this.lastChild) {
+      this.remove(this.lastChild);
+    }
+
+    const tempA = document.createElement('a');
+    tempA.href = panel.config.html_url || panel.config.js_url;
+
+    if (!panel.config.trust_external && !['localhost', '127.0.0.1', location.hostname].includes(tempA.hostname)) {
+      if (!confirm(`Do you trust the external panel "${panel.config.name}" at "${tempA.href}"?
```

Comment:
Afaik confirm does not work on mobile Safari or the iOS App

Reply:
https://stackoverflow.com/questions/18903525/alert-and-confirm-not-working-with-apple-mobile-web-app-capable

Fixed in iOS 7.0.3 according to this StackOverflow.